presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduModule.java

```diff
@@ -50,14 +51,16 @@ protected void configure()
     {
         bind(TypeManager.class).toInstance(typeManager);
 
-        bind(KuduConnector.class).in(Scopes.SINGLETON);
+        bind(KuduSessionProperties.class).in(Scopes.SINGLETON);
+
         bind(KuduMetadata.class).in(Scopes.SINGLETON);
         bind(KuduTableProperties.class).in(Scopes.SINGLETON);
         bind(ConnectorSplitManager.class).to(KuduSplitManager.class).in(Scopes.SINGLETON);
         bind(ConnectorPageSourceProvider.class).to(KuduPageSourceProvider.class)
                 .in(Scopes.SINGLETON);
         bind(ConnectorPageSinkProvider.class).to(KuduPageSinkProvider.class).in(Scopes.SINGLETON);
         bind(KuduHandleResolver.class).in(Scopes.SINGLETON);
+        bind(ConnectorNodePartitioningProvider.class).to(KuduNodePartitioningProvider.class).in(Scopes.SINGLETON);
```

Comment:
```suggestion
        bind(ConnectorNodePartitioningProvider.class).annotatedWith(ForClassLoaderSafe.class).to(KuduNodePartitioningProvider.class).in(Scopes.SINGLETON);
        bind(ConnectorNodePartitioningProvider.class).to(ClassLoaderSafeNodePartitioningProvider.class).in(Scopes.SINGLETON);
```

This ensures that we can directly Inject KuduConnector